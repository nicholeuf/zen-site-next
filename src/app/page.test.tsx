/*
 * @jest-environment jsdom
 */
import {
  renderWithLayout,
  screen,
  renderSnapshotWithLayout,
  resetMatchMedia,
  XS_DEVICE,
  MD_DEVICE,
  waitFor,
  resolvedComponent,
} from 'test-utils';
import userEvent from '@testing-library/user-event';

import HomePage from './page';

const mockUsePathname = jest.fn();
const mockGetPlaceholderImage = jest.fn();

jest.mock('next/navigation', () => ({
  usePathname() {
    return mockUsePathname();
  },
}));

jest.mock('./lib/getPlaceholderImage', () => () => mockGetPlaceholderImage());

describe('The Home Page', () => {
  beforeAll(() => {
    mockUsePathname.mockImplementation(() => '/');
    mockGetPlaceholderImage.mockImplementation(() => 'blurred-image');
    resetMatchMedia();
  });

  test('has expected snapshot', async () => {
    const HomeResolved = await resolvedComponent(HomePage);
    const component = renderSnapshotWithLayout(<HomeResolved />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('User with MD Device', () => {
    beforeAll(() => {
      resetMatchMedia(MD_DEVICE);
    });

    test('can interact with navigation', async () => {
      const HomeResolved = await resolvedComponent(HomePage);
      renderWithLayout(<HomeResolved />);

      // The home link in the top left corner
      expect(
        screen.getAllByRole('link', {
          name: 'Home',
        })[0]
      ).toBeVisible();

      expect(
        screen.getByRole('navigation', {
          name: 'Main Navigation',
        })
      ).toBeVisible();

      const workLink = screen.getByRole('link', {
        name: 'Work',
      });
      expect(workLink).not.toHaveAttribute('aria-current', 'page');
      const homeLink = screen.getAllByRole('link', {
        name: 'Home',
      })[1];
      expect(homeLink).toHaveAttribute('aria-current', 'page');
    });
  });

  describe('User with XS Device', () => {
    beforeAll(() => {
      resetMatchMedia(XS_DEVICE);
    });

    test('can interact with mobile navigation', async () => {
      const HomeResolved = await resolvedComponent(HomePage);
      renderWithLayout(<HomeResolved />);

      const user = userEvent.setup();
      const getDialog = () => screen.queryByTestId('mobile-navigation-dialog');

      // Dialog is not initially present in document
      expect(getDialog()).not.toBeInTheDocument();

      const openButton = screen.getByRole('button', {
        name: 'Mobile Navigation Trigger',
      });
      await waitFor(() => expect(openButton).toHaveFocus());
      expect(openButton).toHaveAttribute('aria-expanded', 'false');

      // Clicking the open button opens the dialog
      await user.click(openButton);
      expect(getDialog()).toBeVisible();
      expect(openButton).toHaveAttribute('aria-expanded', 'true');

      const closeButton = screen.getByRole('button', {
        name: 'Close Mobile Navigation',
      });
      await waitFor(() => expect(closeButton).toHaveFocus());

      expect(
        screen.getByRole('navigation', {
          name: 'Main Navigation',
        })
      ).toBeVisible();

      const workLink = screen.getByRole('link', {
        name: 'Work',
      });
      expect(workLink).not.toHaveAttribute('aria-current', 'page');
      const homeLink = screen.getByRole('link', {
        name: 'Home',
      });
      expect(homeLink).toHaveAttribute('aria-current', 'page');

      // Clicking the close button closes the dialog
      await user.click(closeButton);
      await waitFor(() => expect(getDialog()).not.toBeInTheDocument());
      expect(openButton).toHaveFocus();
      expect(openButton).toHaveAttribute('aria-expanded', 'false');
    });
  });
});
