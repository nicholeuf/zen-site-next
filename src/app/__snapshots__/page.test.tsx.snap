// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home Page has expected snapshot 1`] = `
[
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: fixed;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
  background-color: #FA2742;
  color: #fff;
  background-color: #E8EAE3;
  border: 8px solid;
  border-color: #373833;
  min-height: 60px;
  height: 60px;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: none;
}

@media print {
  .emotion-0 {
    position: absolute;
  }
}

.emotion-1 {
  font-family: fontFamily;
  font-size: 30px;
  height: 100%;
  width: 60px;
  border-right: 8px solid;
  border-right-color: #373833;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1:hover {
  cursor: pointer;
}

.emotion-2 {
  margin: 0;
  font: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #373833;
}

.emotion-2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:hover {
  color: rgb(251, 82, 103);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-transition: -webkit-transform 0.25s ease;
  transition: transform 0.25s ease;
}

.emotion-4.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-4.Mui-selected {
  background-color: rgba(250, 39, 66, 0.08);
}

.emotion-4.Mui-selected.Mui-focusVisible {
  background-color: rgba(250, 39, 66, 0.2);
}

.emotion-4.Mui-disabled {
  opacity: 0.38;
}

.emotion-4:hover {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.emotion-5 {
  margin: 0;
  font: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  letter-spacing: 2.75px;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #373833;
  -webkit-transition: color 0.25s ease;
  transition: color 0.25s ease;
}

.emotion-5:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-5:hover {
  color: rgb(251, 82, 103);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed emotion-0"
    data-testid="header"
  >
    <div
      className="MuiBox-root emotion-1"
      data-testid="header-logo"
    >
      <a
        aria-label="Home"
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-2"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        nf
      </a>
    </div>
    <nav
      data-testid="header-nav"
    >
      <ul
        className="MuiList-root MuiList-padding emotion-3"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
          disabled={false}
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-5"
            href="/work"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Work
          </a>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
          disabled={false}
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-5"
            href="/about"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-4"
          disabled={false}
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-5"
            href="/contact"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  margin-top: 60px;
  min-height: calc(100vh - 60px - 160px);
  padding: 0!important;
  max-width: 100%!important;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  min-height: calc(100vh - 60px - 160px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  position: relative;
}

.emotion-2 {
  height: 100%;
  width: 100%;
  background: url(https://res.cloudinary.com/nicholeuf/image/upload/o_30/f_auto/q_5/v1/zensite/scott-webb-5mIcH3q7tIk-unsplash?_a=BAVAfVBy0) no-repeat;
  -webkit-background-position: 50% 25%;
  background-position: 50% 25%;
  -webkit-background-size: cover;
  background-size: cover;
  position: absolute;
  bottom: 0;
  right: 0;
  z-index: -1;
}

.emotion-3 {
  margin-top: 24px;
  margin-bottom: 24px;
  display: grid;
  gap: 16px;
}

@media (min-width:0px) {
  .emotion-3 {
    grid-template-areas: "photo" "heading" "body";
  }
}

@media (min-width:600px) {
  .emotion-3 {
    grid-template-areas: "photo heading heading heading" "body body body body";
  }
}

.emotion-4 {
  grid-area: photo;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  grid-area: heading;
}

.emotion-6 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 300;
  font-size: 3.5rem;
  line-height: 1.167;
  margin-bottom: 0.35em;
}

@media (min-width:600px) {
  .emotion-6 {
    font-size: 4.7129rem;
  }
}

@media (min-width:900px) {
  .emotion-6 {
    font-size: 5.3556rem;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    font-size: 5.9983rem;
  }
}

.emotion-7 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1.5625rem;
  line-height: 1.235;
  margin-bottom: 0.35em;
  font-weight: 500;
}

@media (min-width:600px) {
  .emotion-7 {
    font-size: 1.8219rem;
  }
}

@media (min-width:900px) {
  .emotion-7 {
    font-size: 2.0243rem;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    font-size: 2.0243rem;
  }
}

.emotion-8 {
  grid-area: body;
}

.emotion-9 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  max-width: 900px;
  color: #424242;
}

@media (min-width:0px) {
  .emotion-9 {
    font-size: 1.25rem;
  }
}

@media (min-width:600px) {
  .emotion-9 {
    font-size: 1.5rem;
  }
}

<main
    className="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      className="MuiBox-root emotion-1"
      data-testid="landing"
    >
      <div
        className="MuiBox-root emotion-2"
        data-testid="landing-background-image"
      />
      <div
        className="MuiBox-root emotion-3"
      >
        <div
          className="MuiBox-root emotion-4"
        >
          <img
            alt="Photo of website author"
            data-nimg="1"
            data-testid="landing-profile-image"
            decoding="async"
            height={200}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://res.cloudinary.com/nicholeuf/image/upload/c_thumb,w_200,h_200,g_auto,z_0.5/r_20/f_auto/q_auto/v1/zensite/nf-profile-c-hibiscus?_a=BAVAfVBy0"
            srcSet="https://res.cloudinary.com/nicholeuf/image/upload/c_thumb,w_200,h_200,g_auto,z_0.5/r_20/f_auto/q_auto/v1/zensite/nf-profile-c-hibiscus?_a=BAVAfVBy0 1x, https://res.cloudinary.com/nicholeuf/image/upload/c_thumb,w_200,h_200,g_auto,z_0.5/r_20/f_auto/q_auto/v1/zensite/nf-profile-c-hibiscus?_a=BAVAfVBy0 2x"
            style={
              {
                "border": "1px solid white",
                "borderRadius": "20px",
                "color": "transparent",
              }
            }
            width={200}
          />
        </div>
        <div
          className="MuiBox-root emotion-5"
        >
          <h1
            className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom emotion-6"
          >
            Hi, I'm Nichole
          </h1>
          <h2
            className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom emotion-7"
          >
            Full-Stack Developer
          </h2>
        </div>
        <div
          className="MuiBox-root emotion-8"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 emotion-9"
          >
            Based in Orlando, Florida, my passion lies in crafting seamless, intuitive, well-built experiences for the customer. I've mentored junior developers, acted as a Lead Front-End developer, and collaborated within dynamic, agile teams.
          </p>
        </div>
      </div>
    </div>
  </main>,
  .emotion-0 {
  background-color: #373833;
  color: #E8EAE3;
  height: 160px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-1 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-transition: -webkit-transform 0.25s ease;
  transition: transform 0.25s ease;
}

.emotion-2.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-2.Mui-selected {
  background-color: rgba(250, 39, 66, 0.08);
}

.emotion-2.Mui-selected.Mui-focusVisible {
  background-color: rgba(250, 39, 66, 0.2);
}

.emotion-2.Mui-disabled {
  opacity: 0.38;
}

.emotion-2:hover {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.emotion-3 {
  margin: 0;
  font: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10 {
  text-align: center;
}

.emotion-11 {
  margin: 0;
  font-family: fontFamily;
  font-size: 26px;
}

.emotion-12 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  display: inline;
  color: #FA2742;
}

.emotion-13 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 12px;
}

.emotion-14 {
  margin: 0;
  font: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #E8EAE3;
  font-size: 12px;
}

.emotion-14:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<footer
    className="MuiBox-root emotion-0"
    data-testid="footer"
  >
    <nav>
      <ul
        className="MuiList-root MuiList-padding emotion-1"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-2"
          disabled={false}
        >
          <a
            aria-label="LinkedIn"
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-3"
            href="https://www.linkedin.com/in/nicholeuf"
            onBlur={[Function]}
            onFocus={[Function]}
            rel="noopener"
            target="_blank"
          >
            <img
              alt="LinkedIn External Link"
              data-nimg="1"
              decoding="async"
              height={24}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Flinkedin%2FE8EAE3&w=48&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Flinkedin%2FE8EAE3&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Flinkedin%2FE8EAE3&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={24}
            />
          </a>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-2"
          disabled={false}
        >
          <a
            aria-label="GitHub"
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-3"
            href="https://github.com/nicholeuf"
            onBlur={[Function]}
            onFocus={[Function]}
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Github External Link"
              data-nimg="1"
              decoding="async"
              height={24}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fgithub%2FE8EAE3&w=48&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fgithub%2FE8EAE3&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fgithub%2FE8EAE3&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={24}
            />
          </a>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-2"
          disabled={false}
        >
          <a
            aria-label="NPM"
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-3"
            href="https://www.npmjs.com/~nicholeuf"
            onBlur={[Function]}
            onFocus={[Function]}
            rel="noopener"
            target="_blank"
          >
            <img
              alt="NPM External Link"
              data-nimg="1"
              decoding="async"
              height={24}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fnpm%2FE8EAE3&w=48&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fnpm%2FE8EAE3&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fnpm%2FE8EAE3&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={24}
            />
          </a>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-2"
          disabled={false}
        >
          <a
            aria-label="CodePen"
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-3"
            href="https://codepen.io/nicholeuf-the-vuer"
            onBlur={[Function]}
            onFocus={[Function]}
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Codepen External Link"
              data-nimg="1"
              decoding="async"
              height={24}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fcodepen%2FE8EAE3&w=48&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fcodepen%2FE8EAE3&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.simpleicons.org%2Fcodepen%2FE8EAE3&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={24}
            />
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="MuiBox-root emotion-10"
    >
      <p
        className="MuiTypography-root MuiTypography-sacramento emotion-11"
      >
        Made with
         
        <span
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          ♥
        </span>
         
        by Nichole Frey
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 emotion-13"
      >
        Copyright © 
        2024
         -
         
        <strong>
          All Rights Reserved
        </strong>
         -
         
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-14"
          href="https://github.com/nicholeuf/zen-site-next"
          onBlur={[Function]}
          onFocus={[Function]}
          rel="noopener"
          target="_blank"
        >
          [view source code]
        </a>
      </p>
    </div>
  </footer>,
]
`;
